draft-ietf-add-split-horizon-authority.xml

```diff
@@ -179,158 +185,163 @@
+        <li>Let $TOKEN = IH($SALT, $X, 0)</li>
+      </ol>
+      <t>The zone operator then publishes a "Verification Record" with the
+      following structure, following the advice of
+      <xref target="I-D.ietf-dnsop-domain-verification-techniques"/>:</t>
+      <ul>
+        <li>Type = TXT.</li>
+        <li>Owner Name = Concatenation of the ADN, "_splitdns-challenge", and
+            the parent zone name.</li>
+        <li>Contents = "token=base64url($TOKEN)".</li>
+      </ul>
+      <t>By publishing this record, the parent zone authorizes the local
+      encrypted resolver to serve these subdomains authoritatively.</t>
+      <section>
+        <name>Example</name>
+        <t>Consider the following authorization claim:</t>
+        <ul>
+          <li>ADN = "resolver17.parent.example"</li>
+          <li>Parent = "parent.example"</li>
+          <li>Subdomains = "payroll.parent.example",
+              "secret.project.parent.example"</li>
+          <li>Hash Algorithm = SHA-1</li>
+          <li>Salt = "example salt bytes (should be random)"</li>
+        </ul>
+        <t>To approve this claim, the zone operator would publish the following record:</t>
+        <figure><artwork><![CDATA[
+resolver17.parent.example._splitdns-challenge.parent.example. IN TXT \
+"token=kfHudyzZkdQccYVLV0zWsY4ETZUK"
+]]></artwork></figure>
+      </section>
+      <section>
+        <name>Conveying Authorization Claims</name>
+        <t>
+          The Authorization Claim is an abstract structure that must be encoded in
+          some concrete syntax in order to convey it from the network to the client.
+          This section defines some encodings of the Authorization Claims.
+        </t>
+        <section>
+          <name>Using DHCP</name>
+          <t>
+            In DHCP, each Authorization Claim is encoded as a DHCP Authentication
+            Option <xref target="RFC3118"/>, using the Protocol value $TBD1, "Split
+            DNS Authentication".  The Algorithm field provides the NSEC3 Hash Algorithm,
+            represented by its registered Value.  The Authentication Information
+            <bcp14>MUST</bcp14> contain the following information, concatenated:</t>
+          <ol>
+            <li>A one-octet "salt length" field.</li>
+            <li>The salt value.</li>
+            <li>The ADN in canonical form.</li>
+            <li>The parent name in canonical form.</li>
+            <li>The $X value defined in <xref target="establishing"/>.</li>
+          </ol>
+        </section>
+        <section>
+          <name>Using Provisioning Domains</name>
+          <t>When using Provisioning Domains, the Authorization Claims are represented
+          by the PvD Additional Information key $TBD2, whose value is a
+          JSON Array.  Each entry in the array <bcp14>MUST</bcp14> be a JSON object
+          with the following structure:</t>
+          <ul>
+            <li>"resolver": The ADN as a dot-separated name.</li>
```

Comment:
I think you mean ADN in canonical form.

Reply:
No, "canonical form" is the wire format, which is not friendly to JSON.